lib/Refactoring/AddExplicitCodableImplementation.cpp

```diff
@@ -12,183 +12,230 @@
 
 #include "RefactoringActions.h"
 #include "Utils.h"
+#include "swift/AST/ProtocolConformance.h"
 
 using namespace swift::refactoring;
 
 namespace {
 class AddCodableContext {
 
   /// Declaration context
-  DeclContext *DC;
+  IterableDeclContext *IDC;
 
-  /// Start location of declaration context brace
-  SourceLoc StartLoc;
+  AddCodableContext(IterableDeclContext *IDC) : IDC(IDC){};
```

Comment:
Could have one for nominals and one for extensions? Would make the llvm_unreachable (which should be assertion IMO) impossible too.